Documentation Revamp

vishal-biyani opened the pull request:
Adding documentation as per new structure and details

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fission/fission/496)
<!-- Reviewable:end -->

soamvasani commented:
Thanks @vishal-biyani. Would you mind putting the structure in the PR description too?

@life1347 @timirahj please take a look and comment if you have any ideas/suggestions around the doc structure.

vishal-biyani commented:
Below structure and basic content are targeted as part of this PR. 
Some areas might be kept in the draft based on content amount/maturity etc.
Requesting your comments on structure:

```
.
├── Introduction (Homepage describing basics)
├── Contributing
│   └── Compilation Fission
│   └── Testing Debugging & known issues
├── Installation
│   └── Quick Installation
│   └── Pre-requisites
│   └── Minikube Installation
│   └── Hosted Kubernetes Cluster
│   └── First example
│   └── Limitations and known issues
│   └── Upgrading from older versions
│       └── UPGRADING FROM V0.1 TO V0.2.X
│       └── UPGRADING FROM V0.3 TO V0.4.X
│       └── UPGRADING FROM V0.4 TO V0.5.X
├── Fission Concepts
│   ├── Functions
│   ├── Environments
│       └── Creating custom environments
│   └── Executors 
│   └── Triggers
│       └── Time triggers
│       └── Http Triggers
│       └── MQ Triggers
│   ├── Watches
│   └── Package
├── Using fissoon
│   ├── Functions
│   ├── Environments
│       └── Creating custom environments
│   └── Executor Types
│       └── PoolManager
│       └── NewDeploy
│   └── Triggers
│       └── Time triggers
│       └── Http Triggers
│       └── MQ Triggers
│   ├── Watches
│   └── Package
├── Fission Workflows
│   └── Details of Workflow (To be further expanded)
└── Resources
    └── Links etc.
```

erwinvaneyk commented on Documentation/docs-site/content/concepts/executor.en.md:
```diff
@@ -0,0 +1,6 @@
+---
+title: "Executor"
+date: 2017-09-07T20:10:05-07:00
```
Date seems to be off in many of the documents, I would propose to just set it to today.

vishal-biyani replied:
Yes, I am not sure if Hugo has a better way, adding TS manually seems odd to me

soamvasani replied:
Date doesn't really matter. We're not displaying it.  You could probably remove it.

erwinvaneyk commented on Documentation/docs-site/content/contributing/_index.en.md:
```diff
@@ -0,0 +1,6 @@
+date: 2017-09-07T20:10:05-07:00
+draft: false
+weight: 50
+---
```
As a starting point, you could just copy and paste the contributing.md that we have in the root

vishal-biyani commented:
@erwinvaneyk - yes still WIP and will add another batch tomorrow for review.

erwinvaneyk commented:
Ok understood, @vishal-biyani :) It might be an idea to have a "wip" label for these kinds of PRs

soamvasani commented on Documentation/docs-site/content/usage/executor.en.md:
```diff
@@ -0,0 +1,6 @@
+---
+title: "Executor"
```
We shouldn't really write a usage guide to the executor; it's a component of our implementation. We should write a usage guide to controlling function execution, latency, throughput, memory costs etc.

soamvasani commented on Documentation/docs-site/content/usage/watch.en.md:
```diff
@@ -0,0 +1,6 @@
+---
+title: "Watch"
```
Same comment as above, watches are just a type of trigger

soamvasani commented on Documentation/docs-site/content/concepts/environments.en.md:
```diff
@@ -0,0 +1,25 @@
+---
+title: "Environemnt"
```
typo

vishal-biyani commented on Documentation/docs-site/content/resources/_index.en.md:
```diff
@@ -0,0 +1,6 @@
+---
```
This is a placeholder for links, blogs and anything which does not fit in the docs currently

soamvasani replied:
Right, but I'm concerned that it will become disorganized. The blog has it's own site.  I suggest removing this section

vishal-biyani commented on Documentation/docs-site/content/installation/_index.en.md:
```diff
@@ -1,181 +1,6 @@
 date: 2017-09-07T20:10:05-07:00
 draft: false
 weight: 20
----
```
On second thoughts, I agree. It is not friendly to have jump n number of pages to install something. Will put in a single one.

vishal-biyani commented on Documentation/docs-site/content/installation/_index.en.md:
```diff
@@ -1,181 +1,6 @@
 ---
-title: "Installation Guide"
```
I am sorry, I have too many coding tendencies while also typing normal text :)  and I like camel case 😈

soamvasani replied:
Oh, I didn't mean the casing of the word guide/Guide. I meant, the diff removes the word guide and changes "Installation Guide" to just "Installation", and I was wondering why you did that.

soamvasani commented on Documentation/docs-site/content/contributing/_index.en.md:
```diff
@@ -0,0 +1,10 @@
+
+# Contributing to Fission
+
+### Development guide
```
link

soamvasani commented on Documentation/docs-site/content/installation/_index.en.md:
```diff
@@ -1,181 +1,10 @@
-for some example functions.
+# Installation
 
+### Installing and upgrading Fission
```
link

soamvasani commented on Documentation/docs-site/content/usage/_index.en.md:
```diff
@@ -0,0 +1,10 @@
+
+# Using Fission
+
+### Examples, code samples and How To's
```
Where does this show up?  Also, suggest rephrasing to "Usage guides, tutorials and examples".

vishal-biyani replied:
So this is a "chapter" title and it shows up like this: 
![image](https://user-images.githubusercontent.com/1167948/36463057-58beda8e-16ed-11e8-867a-6540bb01ea73.png)

soamvasani commented on Documentation/docs-site/content/usage/access-secret-cfgmap-in-function.md:
```diff
@@ -1,8 +1,11 @@
 ---
-title: "Access secret/configmap in function"
+title: "Secret/configmap in function"
```
I liked the previous title better (though "Accessing secrets..." is better than "Access secret...")

vishal-biyani replied:
Yeah, I think I shortened it because it was not looking good, most of the titles are short and this is much longer than others. For now, I changed it back, but I want to get back to aesthetics of this later :)
![image](https://user-images.githubusercontent.com/1167948/36521907-7ff2b1ac-17bf-11e8-91f6-ef4cd79283e0.png)

soamvasani commented on Documentation/docs-site/content/usage/environments.en.md:
```diff
@@ -0,0 +1,37 @@
+
+### Create an environment
+
+You can create an environment from node-env image and specify the resources such as memory and CPU along with poolsize. The poolsize controls how many warm pods are created initially.
```
"You can create an environment on your cluster from an image for that language.  Optionally, you can specify CPU and memory resource limits.  You can also specify the number of initially pre-warmed pods, which is called the poolsize."

soamvasani commented on Documentation/docs-site/content/usage/environments.en.md:
````diff
@@ -0,0 +1,37 @@
+When you create an environment, you can specify a builder image and builder command which will be used for building from source code. You can override the build command when creating a function. For more details on builder and packages you should check out examples in [Functions](../functions) and [packages](../package)
+
+```
+fission env create --name python --image gcr.io/fission-ci/python-env:test --builder gcr.io/fission-ci/python-env-builder:test
````
We shouldn't be telling people to use our test builds... this guide should point at released images only.

vishal-biyani replied:
Yes, I had not found official builder image when I wrote this, fixed now.

soamvasani commented on Documentation/docs-site/content/usage/executor.en.md:
```diff
@@ -0,0 +1,92 @@
+weight: 43
+---
+
+### Pool executor, low latency
```
In this usage guide we shouldn't talk about the executor in the subheadings, only the use case. E.g.  

"Functions with no cold start overhead"
"Functions with no idle cost"
"Autoscaling"

soamvasani commented on Documentation/docs-site/content/usage/package.en.md:
```diff
@@ -0,0 +1,130 @@
+---
+title: "Package"
```
Needs a better title

vishal-biyani commented on Documentation/docs-site/content/concepts/_index.en.md:
```diff
@@ -0,0 +1,10 @@
+
+# Fission Concepts
+
+#### Understanding Fission terminology and concepts
```
No, this is a chapter title and subtitle, for each chapter, it shows up like this:
![image](https://user-images.githubusercontent.com/1167948/36463081-858c3fa2-16ed-11e8-832f-f2c34e42bcbe.png)

And the chapter has subpages - which user can click on and walk through each topic:
![image](https://user-images.githubusercontent.com/1167948/36463090-97e79f2a-16ed-11e8-9ad3-a027ee036c0a.png)

vishal-biyani commented:
Also addresses #470

life1347 commented on Documentation/docs-site/content/concepts/executor.en.md:
```diff
@@ -0,0 +1,39 @@
+
+Newdeploy executortype can be used for requests with no particular low-latency requirements, such as those invoked asynchronously, minscale can be set to zero. In this case the Kubernetes deployment and other objects will be created on first invocation of the function. Subsequent requests can be served by the same deployment. If there are no requests for certain duration then the idle objects are ...
+
+For requests where latency requirements are stringent, a minscale  greater than zero can beset. This essentially keeps a minscale number of pods ready when you create a function. When the function is invoked, there is no delay since the pod does not have to created. Also minscale ensures that the pods are not cleaned up even if the function is idle. This is great for functions where lower latency ...
```
`a minscale greater than zero can be set`

life1347 commented:
Review status: 0 of 25 files reviewed at latest revision, 27 unresolved discussions.

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 25 at r8](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5P-r8-25:-L6G03Dlg5U9TEpq7HCp:b-96kcw6) ([raw file](https://github.com/fission/fission/blob/c90e2ed0eb9cd7ea2fe3e1719f861e2ed676f690/Documentation/docs-site/content/concepts/executor.en.md#L25)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

`a minscale greater than zero can be set`
</blockquote></details>

there is no delay since the pod does not have `to be created`...? 

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 29 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G0OoK1KfIPTw0p7bk:-L6G0OoLuMuafZCvCXl7:b-jo5jlj) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/executor.en.md#L29)):*
> ```Markdown
> ### The latency vs. idle-cost tradeoff
> 
> The executors allow you as a user to decide between latency and a small idle cost tradeoff. Depending on the need you can choose one of the combinations which is optimal for your use case. In future a more intelligent mechanism dispatch mechanism will enable more complex combinations of executors.
> ```

In future, a more intelligent dispatch mechanism ...

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 39 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G29QSWDQYDTArDdX0:-L6G29QSWDQYDTArDdX1:b82qjhe) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/executor.en.md#L39)):*
> ```Markdown
> ### Autoscaling
> 
> The new deployment based executor provides autoscaling for functions based on CPU currently. You can set the intial and maximum CPU for a function and target CPU at which autoscaling will be trigerred. Autoscaling is useful for workloads where you expect intermittant spikes in workloads. It also enables optimal usage of resources to execute functions, by using a baseline capacity with minimum scale and ability to burst upto maximum scale based on spikes in demand.
> ```

based on CPU usage ...
burst up to ....

---

*[Documentation/docs-site/content/concepts/functions.en.md, line 6 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G2nronuTxnzr-jimn:-L6G2nronuTxnzr-jimo:b-ake26) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/functions.en.md#L6)):*
> ```Markdown
> weight: 31
> ---
> A function is piece of code that will be invoked based on a [trigger](../trigger). The code follows the Fission interface. In practice the function is only an entry point for execution and it can be backed by a larger program or module.
> ```

is a piece of...
In practice, the function...

---

*[Documentation/docs-site/content/concepts/trigger.en.md, line 15 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G958ckrudBFY8uwXA:-L6G958ckrudBFY8uwXB:b-cii67c) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/trigger.en.md#L15)):*
> ```Markdown
> ## Time Trigger
> 
> If you want a function to be called at a periodic frequency then the time triggers are perfect for the use case. Time trigger follow cron like specifications and are invoked based on the cron schedule.
> ```

Time triggers follow 

---

*[Documentation/docs-site/content/concepts/trigger.en.md, line 17 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G98R5Ix-hFVDBkzx4:-L6G98R5Ix-hFVDBkzx5:bm0p8tn) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/trigger.en.md#L17)):*
> ```Markdown
> If you want a function to be called at a periodic frequency then the time triggers are perfect for the use case. Time trigger follow cron like specifications and are invoked based on the cron schedule.
> 
> Time trigger bases invocations are great for running scheduled jobs, periodic cleanup jobs, periodic polling based invocations etc. 
> ```

Time trigger based ...

---

*[Documentation/docs-site/content/concepts/trigger.en.md, line 21 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G9Ni4cXaBQ0pQSHzz:-L6G9Ni5oDtqxuF83d4u:b-mtjy36) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/trigger.en.md#L21)):*
> send a reponse
send a response

> the message in queue
the messages in queue

> Currently nats-streaming and azure-storage-queue are supported message queues supported.
Currently, `nats-streaming` and `azure-storage-queue` are supported.  

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/496)*
<!-- Sent from Reviewable.io -->

vishal-biyani commented:
Review status: 0 of 25 files reviewed at latest revision, 27 unresolved discussions.

---

*[Documentation/docs-site/content/concepts/_index.en.md, line 10 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5N-r5-10:-L6GESfasEBkurynGg9B:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/concepts/_index.en.md#L10)):*
<details><summary><i>Previously, vishal-biyani (Vishal) wrote…</i></summary><blockquote>

No, this is a chapter title and subtitle, for each chapter, it shows up like this:
![image](https://user-images.githubusercontent.com/1167948/36463081-858c3fa2-16ed-11e8-832f-f2c34e42bcbe.png)

And the chapter has subpages - which user can click on and walk through each topic:
![image](https://user-images.githubusercontent.com/1167948/36463090-97e79f2a-16ed-11e8-9ad3-a027ee036c0a.png)


</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/environments.en.md, line 2 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5O-r1-2:-L6GETsa_NHgJmGZQp_U:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/concepts/environments.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

typo
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/environments.en.md, line 24 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5O-r5-24:-L6GEUru2x0nvHIewhTH:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/concepts/environments.en.md#L24)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

you can extend one of the environments in the list, or create...
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 3 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5P-r1-3:-L6GEVpX-M4N19ZTpcij:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/concepts/executor.en.md#L3)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Date doesn't really matter. We're not displaying it.  You could probably remove it.
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 25 at r8](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5P-r8-25:-L6GEoRb7i-h8TAzkT6a:b-896fix) ([raw file](https://github.com/fission/fission/blob/c90e2ed0eb9cd7ea2fe3e1719f861e2ed676f690/Documentation/docs-site/content/concepts/executor.en.md#L25)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

there is no delay since the pod does not have `to be created`...? 
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 29 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G0OoK1KfIPTw0p7bk:-L6GEtthLN8pmOAavP6R:b-896fix) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/executor.en.md#L29)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

In future, a more intelligent dispatch mechanism ...
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/executor.en.md, line 39 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G29QSWDQYDTArDdX0:-L6GF8kAsMfcJ8tIv-sN:b-896fix) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/executor.en.md#L39)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

based on CPU usage ...
burst up to ....
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/functions.en.md, line 6 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G2nronuTxnzr-jimn:-L6GFFHedeHWZDDjS6VF:b-896fix) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/functions.en.md#L6)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

is a piece of...
In practice, the function...
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/trigger.en.md, line 15 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G958ckrudBFY8uwXA:-L6GFJhxyrsgxjdYfokY:b-896fix) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/trigger.en.md#L15)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

Time triggers follow 
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/trigger.en.md, line 17 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G98R5Ix-hFVDBkzx4:-L6GFLJhZkcxT7Aig33c:b-896fix) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/trigger.en.md#L17)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

Time trigger based ...
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/trigger.en.md, line 21 at r9](https://reviewable.io:443/reviews/fission/fission/496#-L6G9Ni4cXaBQ0pQSHzz:-L6GFV6sxTDupOjKiEyR:b-896fix) ([raw file](https://github.com/fission/fission/blob/e8edfd674b46cd2ce0fc0153976b9aae59c10689/Documentation/docs-site/content/concepts/trigger.en.md#L21)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

> send a reponse
send a response

> the message in queue
the messages in queue

> Currently nats-streaming and azure-storage-queue are supported message queues supported.
Currently, `nats-streaming` and `azure-storage-queue` are supported.  
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/contributing/_index.en.md, line 6 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5U-r1-6:-L6GFWIqmX4p-gviTi8U:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/contributing/_index.en.md#L6)):*
<details><summary><i>Previously, erwinvaneyk (Erwin van Eyk) wrote…</i></summary><blockquote>

As a starting point, you could just copy and paste the contributing.md that we have in the root
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/contributing/_index.en.md, line 10 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5U-r5-10:-L6GFWn50X-oy-irFOto:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/contributing/_index.en.md#L10)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

link
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/installation/_index.en.md, line 10 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5W-r5-10:-L6GFXEs4D1Hf9cmXTWX:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/installation/_index.en.md#L10)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

link
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/installation/kubernetessetup.en.md, line 2 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5b-r1-2:-L6GFXzTgEuwkckuNGZ6:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/installation/kubernetessetup.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Hmm, this is actually a Kubernetes install, not Fission install
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/_index.en.md, line 10 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5h-r5-10:-L6GFYhXabhWe12qsIbm:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/usage/_index.en.md#L10)):*
<details><summary><i>Previously, vishal-biyani (Vishal) wrote…</i></summary><blockquote>

So this is a "chapter" title and it shows up like this: 
![image](https://user-images.githubusercontent.com/1167948/36463057-58beda8e-16ed-11e8-867a-6540bb01ea73.png)

</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/access-secret-cfgmap-in-function.md, line 2 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5dz9DbqlIIN7jHatmi-r5-2:-L6GFZMN8OmD-947vLYe:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/usage/access-secret-cfgmap-in-function.md#L2)):*
<details><summary><i>Previously, vishal-biyani (Vishal) wrote…</i></summary><blockquote>

Yeah, I think I shortened it because it was not looking good, most of the titles are short and this is much longer than others. For now, I changed it back, but I want to get back to aesthetics of this later :)
![image](https://user-images.githubusercontent.com/1167948/36521907-7ff2b1ac-17bf-11e8-91f6-ef4cd79283e0.png)

</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/environments.en.md, line 2 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5i-r1-2:-L6GF_CHMC7NxBDR6HbC:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/usage/environments.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

typo (spelling)
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/environments.en.md, line 9 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5i-r5-9:-L6GF_ZoZb87s1x-T2ld:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/usage/environments.en.md#L9)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

"You can create an environment on your cluster from an image for that language.  Optionally, you can specify CPU and memory resource limits.  You can also specify the number of initially pre-warmed pods, which is called the poolsize."
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/environments.en.md, line 22 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5i-r5-22:-L6GFaIAbJBD2ZKT1x4m:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/usage/environments.en.md#L22)):*
<details><summary><i>Previously, vishal-biyani (Vishal) wrote…</i></summary><blockquote>

Yes, I had not found official builder image when I wrote this, fixed now.
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/executor.en.md, line 2 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5j-r1-2:-L6GFbYGxMM5PSEN2PLN:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/usage/executor.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

We shouldn't really write a usage guide to the executor; it's a component of our implementation. We should write a usage guide to controlling function execution, latency, throughput, memory costs etc.
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/executor.en.md, line 7 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5j-r5-7:-L6GFbzyFzq7qZQPh-q1:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/usage/executor.en.md#L7)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

In this usage guide we shouldn't talk about the executor in the subheadings, only the use case. E.g.  

"Functions with no cold start overhead"
"Functions with no idle cost"
"Autoscaling"

</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/package.en.md, line 2 at r5](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5l-r5-2:-L6GFclSMYlKL7J2Odai:b-896fix) ([raw file](https://github.com/fission/fission/blob/7955fe43a0faea10b242d13bdcaee8b49d180746/Documentation/docs-site/content/usage/package.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Needs a better title
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/concepts/watch.en.md, line 2 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5T-r1-2:-L6GFdMwYboZzjB3_HBA:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/concepts/watch.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Suggest removing this; a watch isn't really a separate concept, but just a type of trigger.
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/installation/known-issues.en.md, line 6 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5Z-r1-6:-L6GFdkF1i8RiIdjsXhZ:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/installation/known-issues.en.md#L6)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Kinda vague... remove unless we have something to populate this with
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/resources/_index.en.md, line 1 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5g-r1-1:-L6GFeOl5Wu1Fayl2XLM:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/resources/_index.en.md#L1)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Right, but I'm concerned that it will become disorganized. The blog has it's own site.  I suggest removing this section.
</blockquote></details>

Done.

---

*[Documentation/docs-site/content/usage/watch.en.md, line 2 at r1](https://reviewable.io:443/reviews/fission/fission/496#-L5ENxsbTZ86E9z1NI5n-r1-2:-L6GFfZdhT8_Whu4nyZk:b-896fix) ([raw file](https://github.com/fission/fission/blob/55f8de42fc265192ba8933b8d7de5b1f741d5f12/Documentation/docs-site/content/usage/watch.en.md#L2)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Same comment as above, watches are just a type of trigger
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/496)*
<!-- Sent from Reviewable.io -->

soamvasani commented:
Just a few comments, then good to merge --

---

Reviewed 7 of 39 files at r1, 4 of 17 files at r3, 4 of 22 files at r4, 7 of 11 files at r5, 7 of 8 files at r6, 5 of 6 files at r7, 3 of 3 files at r8, 3 of 3 files at r10.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[Documentation/docs-site/content/concepts/package.en.md, line 7 at r10](https://reviewable.io:443/reviews/fission/fission/496#-L6OgzZUSeAOh8DLhN58:-L6OgzZUSeAOh8DLhN59:bopqe42) ([raw file](https://github.com/fission/fission/blob/c21ae25a5db3885a119fb18ba7285db5af009707/Documentation/docs-site/content/concepts/package.en.md#L7)):*
> ```Markdown
> ---
> 
> Most real world applications are more than a single file of code and typically have dependencies on libraries etc. Packages in fission solve three distinct problems:
> ```

This doc needs to specifically define both archives and packages.

But, later.  Let's merge this and improve it iteratively.

---

*[Documentation/docs-site/content/usage/access-secret-cfgmap-in-function.md, line 6 at r10](https://reviewable.io:443/reviews/fission/fission/496#-L6OhSH-i0gmUMpseNyp:-L6OhSH1LK8UvkM65qur:b-3rj058) ([raw file](https://github.com/fission/fission/blob/c21ae25a5db3885a119fb18ba7285db5af009707/Documentation/docs-site/content/usage/access-secret-cfgmap-in-function.md#L6)):*
> ```Markdown
> weight: 47
> ---
> <!--
> ```

Can you please remove this comment and file an issue listing out the specific things in this doc that you think should be changed?

---

*[Documentation/docs-site/content/usage/trigger.en.md, line 2 at r10](https://reviewable.io:443/reviews/fission/fission/496#-L6OiZcqObUTDQz3UYyk:-L6OiZcqObUTDQz3UYyl:bp5vyv9) ([raw file](https://github.com/fission/fission/blob/c21ae25a5db3885a119fb18ba7285db5af009707/Documentation/docs-site/content/usage/trigger.en.md#L2)):*
> ```Markdown
> ---
> title: "Trigger"
> ```

This is missing K8s watch triggers. 

(Add it later, no need to block this)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/496#-:-L6OieddpN2dFoa0mFRm:b-7meyi9)*
<!-- Sent from Reviewable.io -->

vishal-biyani commented:
Updated #499 for changes to cover later and addressed one comment.
